crates/sui-json-rpc/src/governance_api.rs

```diff
@@ -34,49 +37,100 @@ impl GovernanceReadApi {
             .get_move_objects(owner, &StakedSui::type_())
             .await?)
     }
-    async fn get_delegations(&self, owner: SuiAddress) -> Result<Vec<Delegation>, Error> {
-        Ok(self
-            .state
-            .get_move_objects(owner, &Delegation::type_())
-            .await?)
+
+    async fn get_delegated_stakes(&self, owner: SuiAddress) -> Result<Vec<DelegatedStake>, Error> {
```

Comment:
I am trying to understand the delta between the previous return type vs the current return type:

1. Previously the return type was "flat": a list of `DelegatedStake` where there could be multiple `DelegatedStake` for the same validator.
2. This PR aggregates the result by the validator address. It still returns a list of `DelegatedStake`, but there will only be exactly one `DelegatedStake` for the same validator. Each `DelegatedStake` now contains a list of `Delegation` objects that share the same validator.

This change makes it a bit involved to change since there are a lot of places in the wallet code that has strong assumptions([example](https://github.com/MystenLabs/sui/blob/main/apps/wallet/src/ui/app/staking/stake/StakingCard.tsx#L79)) on this and it would take us some time to fully understand the wallet logic and make the right changes.

cc @Jibz-Mysten @Jordan-Mysten it would be great if we could get some help from the FE team to advise on these changes. Thanks!

Reply:
@666lcz I think these changes are fine. I am working on staking changes this week, I can make this update on the wallet site.